runtime-decompiler/src/main/java/org/jrd/backend/data/ArchiveManager.java

```diff
@@ -6,19 +6,33 @@
 import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
 
 public class ArchiveManager {
-    final File c;
+    private static ArchiveManager singleton = null;
+
+    public static ArchiveManager getInstance() {
```

Comment:
note that this is not thread safe. For parameter-less singletons, use holde rpattern with inner class

Reply:
pls wrap the getInstance into holder class!